### ISSUE TITLE
False positive `duplicate-bases` with `TypedDict`

### ISSUE THREAD

**Issue by pettinen** [pylint-dev/pylint#9222]
### Bug description

```python
# pylint: disable=missing-class-docstring,missing-module-docstring

import typing


class A(typing.TypedDict):
    a: int


class B(typing.TypedDict):
    b: int


class ABC(A, B):
    c: int


class ABCD(ABC):
    d: int
```

### Command used

```shell
pylint mre.py
```

### Pylint output

```python
************* Module mre
mre.py:23:0: E0241: Duplicate bases for class 'ABCD' (duplicate-bases)

-----------------------------------
Your code has been rated at 4.44/10
```

### Expected behavior

Should not report an error. No errors if `TypedDict` is inherited with unqualified name, i.e.
```python
from typing import TypedDict


class A(TypedDict):
    a: int


class B(TypedDict):
    b: int

# etc.
```

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.5 (main, Sep  2 2023, 14:16:33) [GCC 13.2.1 20230801]
```

### OS / Environment

Arch Linux inside WSL

**Comment by eli-schwartz** [comment]
Strange that it complains about the fourth class instead of the third class. I was pulling my hair out over this because even as a false positive this is pretty confusing. The line the warning points to has only one parent class...
